src/libraries/Native/Unix/System.Globalization.Native/pal_timeZoneInfo.c

```diff
@@ -22,20 +28,194 @@ ResultCode GlobalizationNative_GetTimeZoneDisplayName(const UChar* localeName,
+            // such that they are grouped together when combined with their base offset, location
+            // and sorted alphabetically.  For example:
+            //
+            //     (UTC-07:00) Mountain Time (Denver)
+            //     (UTC-07:00) Mountain Time (Phoenix)
+            //
+            // Without modification, they would show as:
+            //
+            //     (UTC-07:00) Mountain Standard Time (Phoenix)
+            //     (UTC-07:00) Mountain Time (Denver)
+            //
+            // When combined with the rest of the time zones, having them not grouped together
+            // makes it harder to locate the correct time zone from a list.
+            //
+            // The reason we get the standard name is because TR35 (LDML) defines a rule that
+            // states that metazone generic names should use standard names if there is no DST
+            // transition within a +/- 184 day range near the timestamp being translated.
+            //
+            // See the "Type Fallback" section in:
+            // https://www.unicode.org/reports/tr35/tr35-dates.html#Using_Time_Zone_Names
+            //
+            // This might make sense when attached to an exact timestamp, but doesn't work well
+            // when using the generic name to pick a time zone from a list.
+            // Note that this test only happens when the generic name comes from a metazone.
+            //
+            // ICU implements this test in TZGNCore::formatGenericNonLocationName in
+            // https://github.com/unicode-org/icu/blob/master/icu4c/source/i18n/tzgnames.cpp
+            // (Note the kDstCheckRange 184-day constant.)
+            //
+            // The rest of the code below is a workaround for this issue.  When the generic
+            // name and standard name match, we search through the other time zones for one
+            // having the same base offset and standard name but a shorter generic name.
+            // That will at least keep them grouped together, though note that if there aren't
+            // any found that means all of them are using the standard name.
+            //
+            // TODO: If ICU ever adds an API to get a generic name that doesn't perform the
+            // 184-day check on metazone names, then test for the existence of that new API
+            // and use that instead of this workaround.  Keep the workaround for when the
+            // new API is not available.
+
+            // Get the standard name for this time zone.
+            UChar* standardName = (UChar*)calloc((size_t)resultLength, sizeof(UChar));
+            UCalendar* calendar = ucal_open(timeZoneId, -1, locale, UCAL_DEFAULT, &err);
+            ucal_getTimeZoneDisplayName(calendar, UCAL_STANDARD, locale, standardName, resultLength, &err);
+
+            // See if the generic name is the same as the standard name.
+            if (0 == u_strcmp(result, standardName))
```

Comment:
>(0 == u_strcmp(result, standardName) [](start = 15, length = 36)

nit: for readability, please use instead the style 
(u_strcmp(result, standardName) == 0)